Dungeoneer/src/com/interrupt/dungeoneer/statuseffects/BurningEffect.java

```diff
@@ -25,67 +51,41 @@ public BurningEffect(int time, int damageTimer, int damage) {
 		this.damage = damage;
 		this.statusEffectType = StatusEffectType.BURNING;
 	}
-	
+
 	@Override
-	public void doTick(Actor owner, float delta) { 
+	public void doTick(Actor owner, float delta) {
 		dtimer += delta;
 
-		if(dtimer > damageTimer) {
+		if (dtimer > damageTimer) {
 			dtimer = 0;
 			owner.takeDamage(damage, DamageType.PHYSICAL, null);
-			doFireEffect(owner);
-			Audio.playPositionedSound("mg_pass_poison.mp3", new Vector3(owner.x,owner.y,owner.z), 0.5f, 6f);
+			this.doFireEffect(owner);
+			Audio.playPositionedSound(burnSound, new Vector3(owner.x,owner.y,owner.z), soundVolume, soundRange);
 		}
 
-		// put out the fire, if this is water
-		if(owner.inwater) {
-			Tile waterTile = Game.instance.level.findWaterTile(owner.x, owner.y, owner.z, owner.collision);
-			if(waterTile != null && waterTile.data != null && (waterTile.data.hurts <= 0 || waterTile.data.damageType != DamageType.FIRE)) {
-				active = false;
-			}
-		}
+		if (isEntityInWater()){
+            this.active = false;
+        }
 	}
 
 	public void doFireEffect(Entity owner) {
 		if (!this.showParticleEffect) {
 			return;
 		}
 
-		for(int i = 0; i < 8; i++) {
-			float spreadMod = 1f;
-
-			if (owner instanceof Player) {
-				spreadMod = 2.75f;
-			}
-
-			float scale = 0.5f;
-			Particle p = CachePools.getParticle();
-			p.tex = 64;
-			p.lifetime = 90;
-			p.scale = scale;
-			p.startScale = 1.0f;
-			p.endScale = 0.125f;
-			p.fullbrite = true;
-			p.checkCollision = false;
-			p.floating = true;
-			p.x = owner.x + (Game.rand.nextFloat() * scale - (scale * 0.5f)) * spreadMod;
-			p.y = owner.y + (Game.rand.nextFloat() * scale - (scale * 0.5f)) * spreadMod;
-			p.z = owner.z + (Game.rand.nextFloat() * owner.collision.z) - 0.5f;
-			p.playAnimation(64, 69, 40f);
-
-			p.za = Game.rand.nextFloat() * 0.004f + 0.004f;
-
-			Game.GetLevel().SpawnNonCollidingEntity(p);
+		for (int i = 0; i < particleCount; i++) {
+		    this.spawnFireParticle();
 		}
 	}
 
 	@Override
 	public void onStatusBegin(Actor owner) {
-		doFireEffect(owner);
+	    this.calculateSpreadMod();
```

Comment:
Previously, **spreadMod** was calculated per particle, which as a result the calculations were repeated and resulted in the same values. Instead i've moved the calculation to **onStatusBegins**, which calculates spreadMod once at the start.

We can probably keep **spreadMod** as a private member value (even tho it's used in one place) just to prevent unnecessary calculations, due to it being scoped inside **doFireEffect** (contained in a loop inside **doTick**).

```java
private float spreadMod = 1f;
public float baseSpreadMod = 1f;
public float playerSpreadMod = 2.75f;
```

```suggestion
spreadMod = (owner instanceof Player) ? playerSpreadMod : baseSpreadMod;
```